app/jobs/subscription_payment_job.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class SubscriptionPaymentJob < ApplicationJob
+  GRACE_PERIOD = 7.days
+  SUBSCRIPTION_PERIOD = 1.month
+
+  queue_as :default
+
+  def perform(subscription)
+    plan = subscription.plan
+
+    stripe_charge = create_charge(subscription)
+
+    create_donation(subscription, stripe_charge) if stripe_charge
+  end
+
+  protected
+
+  def displayable_amount(amount)
+    return '$0' unless amount
+
+    ActionController::Base.helpers.number_to_currency(amount.to_f / 100)
+  end
+
+  private
+
+  def create_charge(subscription)
+    customer = find_stripe_customer(subscription.user)
+    plan = subscription.plan
+
+    client = Stripe::StripeClient.new
+    charge, resp = client.request do
+      Stripe::Charge.create(
+        customer: customer,
+        amount: (plan.amount * 100).to_i, # amount in cents
+        description: "Charged #{displayable_amount(plan.amount)} for #{plan.name}",
+        currency: 'usd',
+        metadata: { 'plan_id' => plan.id, 'user_id' => subscription.user.id }
+      )
+    end
+
+    p '/' * 80
```

Comment:
@castrolem let's use the Rails logger for this.

Reply:
Actually. This should be deleted.